src/main/kotlin/org/jetbrains/research/testspark/tools/llm/Llm.kt

```diff
@@ -23,132 +18,127 @@ import org.jetbrains.research.testspark.tools.template.Tool
+     * @param fileUrl the URL of the file to generate tests for (optional).
+     * @param testSamplesCode the code of the test samples to use for test generation.
      */
-    override fun generateTestsForMethod(project: Project, psiFile: PsiFile, caretOffset: Int, fileUrl: String?, testSamplesCode: String, testGenerationController: TestGenerationController) {
+    override fun generateTestsForMethod(
+        project: Project,
+        psiHelper: PsiHelper,
+        caretOffset: Int,
+        fileUrl: String?,
+        testSamplesCode: String,
+        testGenerationController: TestGenerationController,
+    ) {
         if (!LLMHelper.isCorrectToken(project, testGenerationController.errorMonitor)) {
             testGenerationController.finished()
             return
         }
-        val psiMethod: PsiMethod = PsiHelperFactory.getPsiHelper(psiFile).getSurroundingMethod(psiFile, caretOffset)!!
-        val codeType = FragmentToTestData(CodeType.METHOD, PsiHelperFactory.getPsiHelper(psiFile).generateMethodDescriptor(psiMethod))
-        createLLMPipeline(project, psiFile, caretOffset, fileUrl, testGenerationController).runTestGeneration(getLLMProcessManager(project, psiFile, caretOffset, testSamplesCode), codeType)
+        val psiMethod = psiHelper.getSurroundingMethod(caretOffset)!!
+        val codeType = FragmentToTestData(CodeType.METHOD, psiHelper.generateMethodDescriptor(psiMethod))
+        createLLMPipeline(project, psiHelper, caretOffset, fileUrl, testGenerationController).runTestGeneration(
+            LLMProcessManager(
+                project,
+                PromptManager(project, psiHelper, caretOffset),
+                testSamplesCode,
+            ),
+            codeType,
+        )
     }
 
     /**
      * Generates tests for a specific line of code.
      *
      * @param project The current project.
-     * @param psiFile The PSI file containing the code.
+     * @param psiHelper the PsiHelper associated with the pipeline.
      * @param caretOffset The offset position of the caret.
      * @param fileUrl The URL of the file.
      * @param testSamplesCode The code for the test samples.
      */
-    override fun generateTestsForLine(project: Project, psiFile: PsiFile, caretOffset: Int, fileUrl: String?, testSamplesCode: String, testGenerationController: TestGenerationController) {
+    override fun generateTestsForLine(
+        project: Project,
+        psiHelper: PsiHelper,
+        caretOffset: Int,
+        fileUrl: String?,
+        testSamplesCode: String,
+        testGenerationController: TestGenerationController,
+    ) {
         if (!LLMHelper.isCorrectToken(project, testGenerationController.errorMonitor)) {
             testGenerationController.finished()
             return
         }
-        val selectedLine: Int = PsiHelperFactory.getPsiHelper(psiFile).getSurroundingLine(psiFile, caretOffset)?.plus(1)!!
+        val selectedLine: Int = psiHelper.getSurroundingLine(caretOffset)?.plus(1)!!
```

Comment:
Why do we add `+1`?

Reply:
@arksap2002 This question is also to you, since this is your part of the code

Reply:
I remember this part vaguely, but I think the offset was starting from 0 and this is why we needed to add 1 to get the actual line number

Reply:
It is better to mention it in a comment above the line.